src/components/common/Swap/components/CoinsListModal/CoinsListModal.tsx

```diff
@@ -54,15 +53,13 @@ const CoinsListModal = ({ selectCoin, balances }: Props) => {
       }
 
       if (balances) {
-        const aDecimals = coinsConfig.get(coinA.name)?.decimals!;
-        const aBalance = balances.find((b) => b.assetId === coinA.assetId)?.amount.toNumber();
-        const aBalanceValue = aBalance ? aBalance / 10 ** aDecimals : 0;
-        const bDecimals = coinsConfig.get(coinB.name)?.decimals!;
-        const bBalance = balances.find((b) => b.assetId === coinB.assetId)?.amount.toNumber();
-        const bBalanceValue = bBalance ? bBalance / 10 ** bDecimals : 0;
-
-        if (bBalanceValue !== aBalanceValue) {
-          return bBalanceValue - aBalanceValue;
+        const aBalance = balances.find((b) => b.assetId === coinA.assetId)?.amount ?? new BN(0);
+        const bBalance = balances.find((b) => b.assetId === coinB.assetId)?.amount ?? new BN(0);
```

Comment:
We need to account for decimals when doing the comparison

Reply:
Fixed